src/main/java/org/jenkinsci/plugins/authorizeproject/strategy/SystemAuthorizationStrategy.java

```diff
@@ -0,0 +1,227 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.authorizeproject.strategy;
+
+import hudson.Extension;
+import hudson.model.Descriptor;
+import hudson.model.Job;
+import hudson.model.Queue;
+import hudson.security.ACL;
+import hudson.util.FormValidation;
+import java.io.IOException;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.acegisecurity.Authentication;
+import org.jenkinsci.plugins.authorizeproject.AuthorizeProjectProperty;
+import org.jenkinsci.plugins.authorizeproject.AuthorizeProjectStrategy;
+import org.jenkinsci.plugins.authorizeproject.AuthorizeProjectStrategyDescriptor;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
+
+/**
+ * Run builds as {@link ACL#SYSTEM}. Using this strategy becomes important when
+ * {@link org.jenkinsci.plugins.authorizeproject.GlobalQueueItemAuthenticator}
+ * is forcing jobs to a user other than {@link ACL#SYSTEM}.
+ *
+ * @since 1.1.1
+ */
+public class SystemAuthorizationStrategy extends AuthorizeProjectStrategy {
+
+    @DataBoundConstructor
+    public SystemAuthorizationStrategy() {
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Authentication authenticate(Job<?, ?> job, Queue.Item item) {
+        return ACL.SYSTEM;
+    }
+
+    /**
+     * Return {@link SystemAuthorizationStrategy} configured in a job.
+     *
+     * @param job the {@link Job}
+     * @return the {@link SystemAuthorizationStrategy} or {@code null}
+     */
+    protected static SystemAuthorizationStrategy getCurrentStrategy(Job<?, ?> job) {
+        if (job == null) {
+            return null;
+        }
+
+        AuthorizeProjectProperty prop = job.getProperty(AuthorizeProjectProperty.class);
+        if (prop == null) {
+            return null;
+        }
+
+        AuthorizeProjectStrategy strategy = prop.getStrategy();
+        if (!(strategy instanceof SystemAuthorizationStrategy)) {
+            return null;
+        }
+
+        return (SystemAuthorizationStrategy) strategy;
+    }
+
+    /**
+     * Called when XSTREAM2 instantiates this from XML configuration.
+     *
+     * When configured via REST/CLI, {@link Descriptor#newInstance(StaplerRequest, JSONObject)} is not called.
+     * Instead checks authentication here.
+     *
+     * @return return myself.
+     * @throws IOException authentication failed.
+     */
+    private Object readResolve() throws IOException {
+        Jenkins instance = Jenkins.getInstance();
+        if (instance == null || !instance.hasPermission(Jenkins.ADMINISTER)) {
```

Comment:
`SpecificUsersAuthorizationStrategy` checks if the user is `SYSTEM` and bombs out otherwise... but that has issues as I see it, in that only `SYSTEM` can then push a job configuration via XML... So I think it is "ok" with its current implementation, but perhaps its current implementation has other bugs